### ISSUE TITLE
Constant Replacement Mutator mutates boolean return values invalidly or equivalently

### ISSUE THREAD

**Issue by Vampire** [hcoles/pitest#658]
The Constant Replacement Mutator behaves somewhat strangely, or should not do some work it currently does.
Spoken in bytecode, it replaces `ICONST_1` by `ICONST_M1` or `ICONST_0` or `ICONST_2` and so on.
But these are used for boolean return values that should be either `0` or `1`.
So I guess in case it is a boolean return value, it should not modify it, or only toggle between those two.

Here is an example of the method mutated in bytecode dump:
```
  // access flags 0x2
  private allowCommandByChannelId(Lorg/javacord/api/entity/message/Message;)Z
   L0
    LINENUMBER 122 L0
    ALOAD 1
    INVOKEINTERFACE org/javacord/api/entity/message/Message.getChannel ()Lorg/javacord/api/entity/channel/TextChannel; (itf)
    INVOKEINTERFACE org/javacord/api/entity/channel/TextChannel.getId ()J (itf)
    ALOAD 0
    GETFIELD net/kautler/command/api/restriction/javacord/ChannelJavacord.channelId : J
    LCMP
    IFNE L1
    ICONST_1
    GOTO L2
   L1
   FRAME FULL [net/kautler/command/api/restriction/javacord/ChannelJavacord org/javacord/api/entity/message/Message] []
    ICONST_0
   L2
   FRAME FULL [net/kautler/command/api/restriction/javacord/ChannelJavacord org/javacord/api/entity/message/Message] [I]
    IRETURN
   L3
    LOCALVARIABLE this Lnet/kautler/command/api/restriction/javacord/ChannelJavacord; L0 L3 0
    LOCALVARIABLE message Lorg/javacord/api/entity/message/Message; L0 L3 1
    MAXSTACK = 4
    MAXLOCALS = 2
```

If it mutates to `ICONST_M1`, the mutant is not killed (seems that `1` and `-1`  are treated equally) .
If it mutates to `ICONST_0`, the mutant is killed, as the method then always returns `false`.
What is also interesting, if it mutates to `ICONST_2`, the mutant is also killed, so maybe the part that interprets the return value on bytecode level does something like `Math.abs(returnValue % 2)` or whatever.

**Comment by hcoles** [comment]
@LaurentTho3 I think this relate to pit rv.

Its a very long time since I looked at it, but I think the inline constant mutator from the main mutator set (disabled by default) has some strategies in place to deal with this.

**Comment by Vampire** [comment]
Yes, according to documentation it is type aware and I also have that one enabled without problems.

**Comment by LaurentTho3** [comment]
The problem here (and in https://github.com/hcoles/pitest/issues/663, https://github.com/hcoles/pitest/issues/661, and https://github.com/hcoles/pitest/issues/662), is that booleans are treated as integers in the bytecode and thus PIT/the operators do not know that they are mutating a boolean. As far as I know there is no way to know if we are mutating a boolean or something else.
The solution adopted in the inline constant mutator is to treat 1 and 0 as special cases and only mutate them to 0 and 1. This works for constant replacement but not for the other operators, and also makes us miss the genuine replacements.
